includes/acl/clip_reader/clip_reader.h

```diff
@@ -0,0 +1,450 @@
+			}))
+			{
+				goto error;
+			}
+
+			bones = allocate_type_array<RigidBone>(m_allocator, m_num_bones);
+
+			m_bone_names = allocate_type_array<char>(m_allocator, name_buffer_length);
+			char* current_name = m_bone_names;
+
+			m_parser.restore_state(before_bones);
+
+			if (!for_each_bone([bones, &current_name](Bone src)
+			{
+				RigidBone* dst = bones + src.index;
+
+				std::memcpy(current_name, src.name, src.name_length);
+				current_name[src.name_length] = '\0';
+				current_name += src.name_length + 1;
+
+				dst->name = nullptr;
+				dst->vertex_distance = src.vertex_distance;
+				dst->bind_rotation = quat_unaligned_load(src.bind_rotation);
+				dst->bind_translation = vector_unaligned_load3(src.bind_translation);
+				// TODO: bind_scale
+
+				return true;
+			}))
+			{
+				goto error;
+			}
+
+			m_parser.restore_state(before_bones);
+
+			if (!for_each_bone([this, bones](Bone src)
+			{
+				if (src.parent_length == 0)
+				{
+					bones[src.index].parent_index = no_parent_bone;
+					return true;
+				}
+
+				int parent_index = index_of_bone(src.parent, src.parent_length);
+				if (parent_index < 0)
+				{
+					set_error(ClipReaderError::NoParentWithThatName);
+					return false;
+				}
+				
+				bones[src.index].parent_index = parent_index;
+				return true;
+			}))
+			{
+				goto error;
+			}
+			
+			m_skeleton = allocate_type<RigidSkeleton>(m_allocator, std::ref(m_allocator), bones, m_num_bones);
+			m_allocator.deallocate(bones);
+
+			return true;
+
+		error:
+			m_allocator.deallocate(bones);
+			return false;
+		}
+
+		int index_of_bone(const char* const name, const int name_length)
+		{
+			const char* bone_name = m_bone_names;
+
+			for (int i = 0; i < m_num_bones; ++i)
+			{
+				if (m_parser.strings_equal(bone_name, name, name_length))
+				{
+					return i;
+				}
+
+				bone_name += std::strlen(bone_name) + 1;
+			}
+
+			return -1;
+		}
+
+		bool for_each_bone(std::function<bool(Bone)> process)
+		{
+			if (!m_parser.array_begins("bones"))
+			{
+				goto error;
+			}
+
+			int index = 0;
+
+			while (!m_parser.try_array_ends())
+			{
+				Bone b;
+
+				if (!m_parser.object_begins() ||
+					!m_parser.read("name", b.name, b.name_length) ||
+					!m_parser.read("parent", b.parent, b.parent_length) ||
+					!m_parser.read("vertex_distance", b.vertex_distance))
+				{
+					goto error;
+				}
+
+				m_parser.try_read("bind_rotation", b.bind_rotation, 4);
+				m_parser.try_read("bind_translation", b.bind_translation, 3);
+				m_parser.try_read("bind_scale", b.bind_scale, 3);
+				
+				if (!m_parser.object_ends())
+				{
+					goto error;
+				}
+
+				b.index = index;
+				++index;
+
+				if (!process(b))
+				{
+					return false;
+				}
+			}
+
+			return true;
+
+		error:
+			m_error = m_parser.get_error();
+			return false;
+		}
+
+		bool create_clip()
+		{
+			m_clip = allocate_type<AnimationClip>(m_allocator, std::ref(m_allocator), std::ref(*m_skeleton), m_header.num_samples, m_header.sample_rate);
```

Comment:
Yeah, it thought I was trying to use the copy constructor and then complained because you disabled it.